python/ray/ml/examples/pytorch/torch_incremental_learning_example.py

```diff
@@ -0,0 +1,597 @@
+  deployment = ModelWrapperDeployment.options(name="mnist_model", route_prefix="/mnist_predict", version=f"v{task_idx}", num_replicas=2)
+  deployment.deploy(batching_params=False, http_adapter=json_to_pandas, predictor_cls=TorchPredictor, checkpoint=latest_checkpoint, model=SimpleMLP(num_classes=10))
+  return deployment.url
+
+# Function that queries our deployed model
+def query_deployment(test_samples: List[np.ndarray], endpoint_uri: str):
+  """Given a set of test samples, queries the model deployment at the provided endpoint and returns the results."""
+  results = []
+  # Have to convert to Python List since Numpy arrays are not Json serializable.
+  for sample in test_samples:
+    results.append(requests.post(endpoint_uri, json={"array": sample.tolist()}))
+  # TODO: Figure out how Serve deals with Pandas DataFrame returned by Predictors.
+  return results
+
+"""# Step 5: Putting it all together
+
+Once we have defined our training logic and our preprocessor, we can put everything together!
+
+For each dataset in our stream, we do the following:
+1. Train on the dataset in Data Parallel fashion. We create a `TorchTrainer`, specify the config for the training loop we defined above, the dataset to train on, and how much we want to scale. `TorchTrainer` also accepts a `checkpoint` arg to continue training from a previously saved checkpoint.
+2. Get the saved checkpoint from the training run.
+3. Test our trained on a test set containing test data from all the tasks trained on so far.
+3. After training on each task, we deploy our model so we can query it for predictions.
+
+In this example, since the training and test data for each task is well-defined beforehand by the benchmark.For real-world scenarios, this probably will not be the case. It is very likely that the prediction requests after training on one task will become the training data for the next task. 
+
+"""
+
+import ray
+from ray.ml.train.integrations.torch import TorchTrainer
+from ray.ml.predictors.integrations.torch import TorchPredictor
+from ray.ml import Checkpoint
+from ray import serve
+from ray.serve.model_wrappers import ModelWrapperDeployment
+from ray.serve.http_adapters import json_to_ndarray
+
+# The number of tasks (i.e. datasets in our stream) that we want to use for this example.
+n_tasks = 3
+
+# Number of epochs to train each task for.
+num_epochs = 4
+# Batch size.
+batch_size = 32
+# Optimizer args.
+learning_rate = 0.001
+momentum = 0.9
+
+# Number of data parallel workers to use for training.
+num_workers = 1
+# Whether to use GPU or not.
+use_gpu = True
+
+permuted_mnist = PermutedMNISTStream(n_tasks=n_tasks)
+train_stream = permuted_mnist.generate_train_stream()
+test_stream = permuted_mnist.generate_test_stream()
+
+latest_checkpoint = None
+
+accuracy_for_all_tasks = []
+task_idx = 0
+all_test_datasets_seen_so_far = []
+for train_dataset, test_dataset in zip(train_stream, test_stream):
```

Comment:
Out of scope for this example, but should we think about how to support the continual learning story better? Currently we basically do

```
checkpoint = None
for ds in datasets:
    trainer = Trainer(..., datasets=ds, resume_from_checkpoint=checkpoint)
    checkpoint = trainer.fit().checkpoint
```

But maybe something like this would be preferable

```
trainer = Trainer()
for ds in datasets:
    trainer.fit(ds)
```

Would be great to collect some user insights on what way is preferable. The main question is if the trainer should retain state after fitting or not - currently it's basically just a dataclass for initialization parameters.

Reply:
Yes 100%. We should definitely do this in a future PR!